src/main/host/syscall/unistd.c

```diff
@@ -297,9 +297,27 @@ SysCallReturn syscallhandler_close(SysCallHandler* sys,
 }
 
 SysCallReturn syscallhandler_dup(SysCallHandler* sys,
-                                   const SysCallArgs* args) {
-    warning("Cannot dup legacy descriptors");
-    return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -EOPNOTSUPP};
+                                 const SysCallArgs* args) {
+    gint fd = args->args[0].as_i64;
+    LegacyDescriptor* desc = process_getRegisteredLegacyDescriptor(sys->process, fd);
+
+    LegacyDescriptorType dType = descriptor_getType(desc);
```

Comment:
What if the plugin passes in a bad FD? I think `desc` will be NULL then, and `descriptor_getType` will fail the magic assertion.

Instead, do we want to return `-EBADF` if `desc` is NULL?

For example, `_syscallhandler_readHelper` does this:

```c
    /* Get the descriptor. */
    LegacyDescriptor* desc = process_getRegisteredLegacyDescriptor(sys->process, fd);
    if (!desc) {
        return (SysCallReturn){.state = SYSCALL_DONE, .retval.as_i64 = -EBADF};
    }
```

Reply:
Thanks, I forgot to check this.